app/controllers/server.rb

```diff
@@ -1,9 +1,91 @@
+require_relative "../models/response.rb"
+require_relative "../models/processor.rb"
+require 'pry'
+
 module RushHour
   class Server < Sinatra::Base
+    include Response, Processor
+
     not_found do
       erb :error
     end
-    
-    # Write the rest of your controller code here!
+
+    get "/" do
+      erb :index
+    end
+
+    get "/sources/:identifier" do |identifier|
+      @client = get_client_stats(identifier)
+      @events = get_client_events(@client) if !@client.nil?
+
+      if Client.find_by(identifier: identifier).nil?
+        @message = "Identifier #{identifier} does not exist!"
+        erb :error
+      elsif Client.find_by(identifier: identifier).payload.empty?
+        @message = "Your identifier #{identifier} does not have any assigned payloads!"
+        erb :error
+      else
+        erb :show
+      end
+    end
+
+    get "/sources/:IDENTIFIER/urls/:RELATIVEPATH" do |identifier, relativepath|
+      @url = get_url_stats("/"+relativepath)
+      @client = get_client_stats(identifier)
+      @events = get_client_events(@client)
+
+      if Url.find_by(path: "/#{relativepath}").nil?
+        @message = "Path #{relativepath} does not exist!"
+        erb :error
+      else
+        erb :show_url
+      end
+    end
+
+    get "/sources/:IDENTIFIER/events/:EVENTNAME" do |identifier, eventname|
+      @eventname = eventname
+      @client = Client.find_by(identifier: identifier)
+      @events = get_client_events(@client)
+      @data = get_event_stats(@client, eventname)
+      @total = @data.values.reduce(:+)
```

Comment:
this is too much logic in the controller. your controller should be asking models or other Ruby objects or modules for data in order to send it to the view